compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala

```diff
@@ -318,7 +318,8 @@ class RefinedPrinter(_ctx: Context) extends PlainPrinter(_ctx) {
         if (name.isTypeName) typeText(txt)
         else txt
       case tree @ Select(qual, name) =>
-        if (qual.isType) toTextLocal(qual) ~ "#" ~ typeText(toText(name))
+        if (tree.symbol == defn.QuotedExpr_~ || tree.symbol == defn.QuotedType_~) keywordStr("~(") ~ toTextLocal(qual) ~ keywordStr(")")
```

Comment:
> This will not work for untyped tree (e.g. -Xprint:parser)

What I meant is that `RefinedPrinter` is used for both typed and untyped tree. You should make sure the tree has a type before accessing its symbol

Reply:
I added the check.